packages/components/src/components/input-date/controler.ts

```diff
@@ -0,0 +1,139 @@
+import { Generic } from '@a11y-ui/core';
+import { Stringified } from '../../types/common';
+import { InputDateType } from '../../types/input/control/number';
+import { Iso8601 } from '../../types/input/iso8601';
+import { InputTypeOnOff } from '../../types/input/types';
+import { watchBoolean, watchJsonArrayString, watchNumber, watchString, watchValidator } from '../../utils/prop.validators';
+import { InputIconController } from '../@deprecated/input/controller-icon';
+import { Props, Watches } from './types';
+
+export class InputDateController extends InputIconController implements Watches {
+	/**
+	 * Regex to check whether a string is a date in ISO-8601 format.
+	 * Test the regex here: https://regex101.com/r/ys3ij0/1
+	 */
+	private readonly isoDateRegex =
+		/(^\d{4}-([0]\d|1[0-2])-([0-2]\d|3[01])([T ][0-2]\d:[0-5]\d:[0-5]\d(?:\.\d+)?([+-][0-2]\d:[0-5]\d|Z)?)?$)|(^[0-2]\d:[0-5]\d(:[0-5]\d)?$)/;
+
+	protected readonly component: Generic.Element.Component & Props;
+
+	public constructor(component: Generic.Element.Component & Props, name: string, host?: HTMLElement) {
+		super(component, name, host);
+		this.component = component;
+	}
+
+	public validateAutoComplete(value?: InputTypeOnOff): void {
+		watchValidator(
+			this.component,
+			'_autoComplete',
+			(value): boolean => typeof value === 'string' && (value === 'on' || value === 'off'),
+			new Set(['on | off']),
+			value
+		);
+	}
+
+	public validateList(value?: Stringified<string[]>): void {
+		watchJsonArrayString(this.component, '_list', (item: string) => typeof item === 'string', value);
+	}
+
+	private readonly parseToString = (value?: Date | string | null) => {
+		if (typeof value === 'string') {
+			return value;
+		}
+		if (typeof value === 'object' && value instanceof Date) {
+			return value.toISOString();
+		}
+		return '';
+	};
+
+	private readonly validateIso8601 = (propName: string, value?: Date | Iso8601 | null, afterPatch?: (v: string) => void) => {
+		return watchValidator(
+			this.component,
+			propName,
+			(value): boolean => value === undefined || value === '' || this.isoDateRegex.test(value),
+			new Set(['Date', 'string{ISO-8601}']),
```

Comment:
The validation is much more complex and depends on the type.

Please reuse the from the component file deleted function `validateDateString` for validation.

Reply:
Moved `validateDateString` to the controller.

Reply:
This function is not used anymore, but I leave it in case it will be helpful.